smart_contract/executor/derive/src/validate.rs

```diff
@@ -37,119 +36,136 @@ enum ValidateAttribute {
     },
 }
 
-impl ValidateAttribute {
-    fn from_attributes<'attr, A>(attributes: A) -> Self
-    where
-        A: IntoIterator<Item = &'attr Attribute>,
-    {
+impl FromAttributes for ValidateAttribute {
+    #[allow(clippy::or_fun_call)] // nope, this is what I actually meant
```

Comment:
Not sure why does this lint need to be disabled?

Reply:
Updated with a comment, duplicating it here:

```rust
    // we use `Option::or` to select the first specified condition in case of duplicates
    // but we still _want_ to validate that each attribute parses successfully
    // this is to ensure that we provide the user with as much validation as possible, instead of bailing out early
    // `Option::or_else` would NOT work here, as it would not validate conditions after the first valid one
```